### ISSUE TITLE
Homebridge hangs and stops responding

### ISSUE THREAD

**Issue by yffud** [iklein99/homebridge-smartthings#257]
https://github.com/iklein99/homebridge-smartthings/blob/44894ec3063e723e757fb4e1608eb9d39c32bb20/src/platform.ts#L73

If the code gets to this catch error it hangs the code and Homebridge until Homebridge is restarted. 

Could there be a retry or continue logic here? 

--
Issue logged with Homebridge https://github.com/homebridge/homebridge/issues/3615
--

Homebridge will stop responding to inputs from HomeKit until it is rebooted.

The last log messages before hanging will be something similar from the Smartthings Plugin

[5/14/2024, 4:33:14 AM] [Smartthings Plug (IK)] Error getting devices from Smartthings: Error: getaddrinfo EAI_AGAIN api.smartthings.com
[5/14/2024, 4:33:14 AM] [Smartthings Plug (IK)] Could not load devices from Smartthings: undefined. Check your configuration

No more long messages after than until rebooted

**Comment by spiazza1** [comment]
Was this issue addressed? It is still affecting v1.5.21.

**Comment by yffud** [comment]
I don’t believe so . I asked ChatGPT to help me write some really hackey retry code to wrap around it to kludge a temporary fix

**Comment by spiazza1** [comment]
I'm going to give this plugin a shot: Restart Plugin
Using the value @daily in order to restart the service once a day.

**Comment by yffud** [comment]
> I'm going to give this plugin a shot: Restart Plugin Using the value @daily in order to restart the service once a day.

Good idea , my first hack was a daily restart of the service by crontab , but sometimes the service was left hung for a few hours then when I needed it
